Fix scraping semester code error

davidatz285 opened the pull request:
1. Merubah koneksi dari halaman frs_prs menjadi ke halaman nilai, karena saya pikir halaman nilai url nya tidak akan berubah seperti halaman frs. Awalnya kode semester diambil dari halaman frs_prs ini. Kalau coba connect ke url studentportal.unpar.ac.id/frs_prs tidak bisa, bisanya ke restu.unpar.ac.id/frs_prs.
2. Mengambil text kode semester dari halaman nilai.

davidatz285 committed:
Fix fetching semester code error

davidatz285 committed:
Fix scraping semester code error

pascalalfadian commented on app/models/support/Scraper.java:
```diff
@@ -1,298 +1,303 @@
-package models.support;
```
Hi David, ini kalau dicompare di website GitHub nya, jadi kayak dia menghapus 300 baris terus menambahkan 300 baris. Penyebabnya karena kamu mengubah line ending nya dari UNIX ke DOS. Tolong dikembalikan line ending nya jadi UNIX lagi atau coba diatur dari git nya: https://www.edwardthomson.com/blog/git_for_windows_line_endings.html

pascalalfadian commented on app/models/support/Scraper.java:
```diff
@@ -1,298 +1,303 @@
+	private final String LOGOUT_URL = BASE_URL + "logout";
+	private final String HOME_URL = BASE_URL + "home";
+	private final String FRSPRS_URL = BASE_URL + "frs_prs";
+	private final String SSO_LOGIN = "https://sso.unpar.ac.id/login?service=https%3A%2F%2Fstudentportal.unpar.ac.id%2FC_home%2Fsso_login";
```
Ini `SSO_LOGIN` kayaknya tidak terpakai di mana-mana ya? Hapus aja.

pascalalfadian commented on app/models/support/Scraper.java:
```diff
@@ -1,298 +1,303 @@
+		   curr_sem = option.val(); 
+		}		
+		curr_sem = curr_sem.substring(2,4).concat(curr_sem.substring(5));
+		System.out.println("cur sem = "+curr_sem);		
```
Debugging output mohon dihapus dari kode.

pascalalfadian commented on app/models/support/Scraper.java:
```diff
@@ -1,298 +1,303 @@
+		doc = resp.parse();	
+		String curr_sem = "";
+		Elements options = doc.getElementsByAttributeValue("name", "dropdownSemester").get(0).children();
+		for (Element option : options) {		    
```
Ini maksudnya mau mengambil elemen terakhir ya? Kalau iya, lebih baik jika ditulis kurang lebih seperti ini biar lebih mudah dipahami:

```
String curr_sem = options.last().val();
```

(dicoba dulu ya soalnya saya nggak compile di sini)

pascalalfadian replied:
@davidatz285 tolong optimize sedikit lagi ya, kode ini kan sebenarnya tidak perlu:

```java
String curr_sem = "";
```

pascalalfadian commented on app/models/support/Scraper.java:
```diff
@@ -1,298 +1,303 @@
+		resp = connection.execute();
+		doc = resp.parse();	
+		String curr_sem = "";
+		Elements options = doc.getElementsByAttributeValue("name", "dropdownSemester").get(0).children();
```
`get(0)` sepertinya bisa diganti jadi `first()`

pascalalfadian reviewed (changes requested):
Hi @davidatz285,

Ada review2 dari saya (ada 5), mohon dicek dan perbaiki ya.

Trims.

davidatz285 committed:
fix scraping semester code error (reviewed)

Kode sudah diperbaiki sesuai review, dan variable yang tidak dipakai sudah dibuang.

davidatz285 committed:
Create .gitattributes

davidatz285 committed:
Update .gitattributes

davidatz285 committed:
Update .gitattributes

davidatz285 committed:
fix scraping semester code error (reviewed)

davidatz285 committed:
Fix scraping semester code error (reviewed)

EOL converted from DOS (CR LF) to UNIX (LF)

pascalalfadian commented:
@davidatz285 ini nanti kalau sudah selesai dari sisi kamu tolong infokan saya ya. Kalau ada kesulitan, juga silakan bertanya di sini. Trims.

davidatz285 commented:
@pascalalfadian oh iya itu kemarin sore saya rasa udh selesai. Maaf tidak mengabari Pak. Di kode commit 
998c7b8  semua yang diminta di review sudah saya kerjakan. Terima kasih dan mohon maaf.

pascalalfadian commented:
Trims. 4 nya udah oke, 1 lagi tolong diresolve ya.

davidatz285 committed:
Optimize method requestNamePhotoTahunSemester

davidatz285 commented:
Sudah diperbaiki sesuai review. Terima kasih.

pascalalfadian commented:
Maaf seperti nya terlewat. Akan saya merge.
